lsp-metals.el

```diff
@@ -114,6 +114,7 @@ more customizations like using environment variables."
   (require 'shr)
   (setq-local show-trailing-whitespace nil)
   (setq-local buffer-read-only nil)
+  (local-set-key (kbd "q") 'kill-this-buffer)
```

Comment:
Should we make this buffer as `special-mode` so the `q` key bind will work automatically? Also that makes buffer read-only too.

Reply:
Your concern is reasonable. It seems to be a overkill in this scenario. I think we don't have any other read only buffer for metals. As for lsp-mode in general, I am not knowledgeable enough.

Reply:
I surely recommend to use existing mode key binds (in this case, `special-mode`) instead of adding new key bind randomly. It allows user customized key binds to work while still provide a good default

Reply:
Yeah. We should do it. What about `(view-mode 1)`?